### ISSUE TITLE
new approach for <h:messages> and to access validation-errors

### ISSUE THREAD

**Issue by gurunrao** [gurunrao/issue_migration#130]
<h:message> and <h:messages> are a good start to place error-messages of
form-validations somewhere, but its behaviour and attributes are very limited in
what you can do with it. I noticed that you can't even define a stylesheet for
the table itself when you set <h:messages layout="table"/>, but only for the
<span>s within the <td>s of the table.

It's hard to explain what one needs to be able to do better and which api is
required to achieve this. I don't know which plans for this might be in queue
for JSF 2.0 and I'm surely not the best to design the perfect api. But I have
attached two screenshots to this issue where you can see which flexibility of
error-messages is prevalent in the web and that JSF is just at the very
beginning in this topic and much more features are needed.

In short these are:

*   a way to check if any validation errors have occured
*   a way to access the error-message(s?) of a certain component-id
*   a way to get a list of all fields that threw a validation-error

This shouldn't be solved by new jsf-tags, but by #

{...}

expressions.

And regarding the table-layout I mentioned at the beginning, it is surely
usefull to add a tableStyle and tableStyleClass-attribut to the api.
#### Environment
Operating System: All
Platform: All
#### Affected Versions
[2.0]

**Comment by gurunrao** [comment]
**Closing this as this issue is migrated to** https://github.com/gurunrao/issue_migration3/issues/130
